common/src/main/java/io/netty5/util/concurrent/FutureCompletionStage.java

```diff
@@ -40,16 +43,109 @@
  *
  * @param <V> the value type.
  */
-public interface FutureCompletionStage<V> extends CompletionStage<V>, java.util.concurrent.Future<V> {
+public interface FutureCompletionStage<V>
+        extends CompletionStage<V>, java.util.concurrent.Future<V>, AsynchronousResult<V> {
 
     /**
-     * Returns the underlying {@link Future} of this {@link FutureCompletionStage}.
+     * Waits for this future until it is done, and rethrows the cause of the failure if this future failed.
+     *
+     * @throws CancellationException if the computation was cancelled
+     * @throws CompletionException   if the computation threw an exception.
+     * @throws InterruptedException  if the current thread was interrupted while waiting
      */
-    Future<V> future();
+    Future<V> sync() throws InterruptedException;
```

Comment:
I think this should return `FutureCompletionStage<V>`.
```suggestion
    FutureCompletionStage <V> sync() throws InterruptedException;
```